shell/browser/mac/electron_application_delegate.mm

```diff
@@ -176,18 +178,10 @@ - (IBAction)newWindowForTab:(id)sender {
 
 - (void)application:(NSApplication*)application
     didRegisterForRemoteNotificationsWithDeviceToken:(NSData*)deviceToken {
-  // https://stackoverflow.com/a/16411517
-  const char* token_data = static_cast<const char*>(deviceToken.bytes);
-  NSMutableString* token_string = [NSMutableString string];
-  for (NSUInteger i = 0; i < deviceToken.length; i++) {
-    [token_string appendFormat:@"%02.2hhX", token_data[i]];
-  }
   // Resolve outstanding APNS promises created during registration attempts
-  electron::api::PushNotifications* push_notifications =
-      electron::api::PushNotifications::Get();
-  if (push_notifications) {
+  if (auto* push_notifications = electron::api::PushNotifications::Get()) {
     push_notifications->ResolveAPNSPromiseSetWithToken(
-        base::SysNSStringToUTF8(token_string));
+        base::HexEncode(electron::util::as_byte_span(deviceToken)));
```

Comment:
(implementation note: the `appendFormat:@"%0.02hhX"`  loop has been replaced with [base::HexEncode](https://chromium.googlesource.com/chromium/src/+/main/base/strings/string_number_conversions.cc#135), which uses [AppendHexEncodedByte()](https://chromium.googlesource.com/chromium/src/+/main/base/strings/string_number_conversions.h#117) which has the same two-uppercase-hex-chars-per-byte behavior as the `%02.2hhX` formatting call)